### PR TITLE
bonzomatic: init at 2018-03-29

### PR DESCRIPTION
**Pull request by ilian**
###### Motivation for this change
Adds [Bonzomatic](https://github.com/Gargaj/Bonzomatic), a live-coding tool for writing 2D fragment/pixel shaders.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

I added the unfreeRedistributable license, as the upstream repo contains [libbass.so](https://www.un4seen.com/bass.html#license) which is only free for non-commercial use.

### CODE REVIEW

**Review by infinisil — APPROVED** [review]
Built and tested this locally, works without problems. It need a simple `config.json` file in the current working directory to get started though. Here's one:

```json
{
  "window":{
    "width":1440,
    "height":900,
    "fullscreen":false
  },
  "font":{
    "file":"/nix/store/h17zrpn0zfgx0p65ic457kb9plwll3n0-nerdfonts-1.1.0/share/fonts/truetype/NerdFonts/Iosevka Nerd Font Complete.ttf",
  }
}
```

(I got the font path with `fc-list`)

**Comment by Infinisil** [comment]
@GrahamcOfBorg build bonzomatic

File: pkgs/applications/editors/bonzomatic/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{ stdenv, makeWrapper, fetchFromGitHub, cmake, alsaLib, mesa_glu, libXcursor, libXinerama, libXrandr, xorgserver }:
+
+let
+  name = "bonzomatic";
+in stdenv.mkDerivation {
+  inherit name;
```
Comment (nlewo) [inline pkgs/applications/editors/bonzomatic/default.nix]: Please add a version in this package (the version is used by nix-env to upgarde). Something such as:
```
name = "${pname}-${version}"
pname = "bonzomatic";
version = "2018-03-29";
```

File: pkgs/applications/editors/bonzomatic/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
+{ stdenv, makeWrapper, fetchFromGitHub, cmake, alsaLib, mesa_glu, libXcursor, libXinerama, libXrandr, xorgserver }:
+
+let
+  name = "bonzomatic";
```
Comment (nlewo) [inline pkgs/applications/editors/bonzomatic/default.nix]: By using a recursive attribute set as derivation argument, this is not required. You can see the `rec` keyword.

**Review by nlewo — CHANGES_REQUESTED** [review]

**Comment by nlewo** [comment]
It would be nice to provide a default configuration file since it is not trivial to create it. This would be useful for reviewers to test future updates. This could be addressed in another PR.
Thanks.

**Pull request merged**